src/api/store/community.py

```diff
@@ -193,19 +198,48 @@ def create_community(self,context: Context, args) -> (dict, MassEnergizeAPIError
         new_action.is_published = False
         new_action.is_global = False
 
-        #make sure the action title does not exceed the max length expected
-        suffix = f'-{new_community.subdomain}-{action_to_copy_id}'
-        new_action.title = new_action.title[:SHORT_STR_LEN-len(suffix)]+suffix 
+        # remove the "TEMPLATE", "TEMP" or "TMP" prefix or suffix
+        old_title = new_action.title
+        new_title = old_title
+
+        loc = old_title.find("TEMPLATE")
+        if loc == 0 :
+          new_title = old_title[9:]
+        elif loc > 0:
+          new_title = old_title[0:loc-1]
+        else:
+          loc = old_title.find("TEMP")
+          if loc==0:
+            new_title = old_title[5:]
+          elif loc>0:
+            new_title = old_title[0:loc-1]
+          else:
+            loc = old_title.find("TMP")
+            if loc==0:
+              new_title = old_title[4:]
+            elif loc>0:
+              new_title = old_title[0:loc-1]
+
+        # add community name to suffix (not the action id
+        # #make sure the action title does not exceed the max length expected
+        #suffix = f'-{new_community.subdomain}-{action_to_copy_id}'
+        suffix = f'-{new_community.subdomain}'
+        len_suffix = len(suffix)
+        new_title_len = min(len(new_title), SHORT_STR_LEN - len_suffix)
+        new_action.title = new_title[0:new_title_len]+suffix
 
         # only save this action if it does not exist
         # if(Action.objects.filter(title=new_action.title, community=new_community).count() == 0):
         new_action.save()
-
         new_action.tags.set(old_tags)
         new_action.vendors.set(old_vendors)
 
         new_action.community = new_community
         new_action.save()
+        num_copied += 1
+        if num_copied >= MAX_TEMPLATE_ACTIONS:
```

Comment:
i think we should not do this.  
I think there is another way to go about what you want to do.  I did the same for Dev.

Basically, in the past there was a bug where while creating a community, in the process of copying the template actions, we were not setting those new actions being created to be is_template=False and so the next time we created something again, those actions showed up again.  It was quite funny when i saw it.  

What i did for dev, which we should do for canary, is to do a backfill instead.  Use the backfill route to delete all nonsense action templates and literally wipe them out.  

And since the bug has been fixed we wont have to deal with that nuisance of creating nonsense actions anymore.  

Good catch though.  I think this the best way to go about it and not necessarily set a bound for number of actions.

Reply:
On the API to just blindly copy hundreds of actions, which is what it was doing, is what I call kamakazi code; leave yourself open to filling the database with junk and hanging the API up in an endless loop.  That is what was happening so this is the fix.  A bound on the number of template actions is a good idea in my opinion; there are 11 now and I could imagine up to 20 but not more than 25.  However- as we get into more regions we may consider having different groups of template actions and at that time it would need to change.